src/app/components/organization-picker.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
+
+import { I18nService } from "jslib-common/abstractions/i18n.service";
+import { OrganizationService } from "jslib-common/abstractions/organization.service";
+import { Utils } from "jslib-common/misc/utils";
+import { Organization } from "jslib-common/models/domain/organization";
+
+@Component({
+  selector: "app-organization-picker",
+  templateUrl: "organization-picker.component.html",
+})
+export class OrganizationPickerComponent implements OnInit {
+  constructor(
+    private organizationService: OrganizationService,
+    private i18nService: I18nService,
+    private route: ActivatedRoute
+  ) {}
+
+  activeOrganizationId: string;
+  organizations: Organization[] = [];
+
+  loaded = false;
+
+  get activeOrganization() {
+    return this.organizations.find((org) => org.id == this.activeOrganizationId);
+  }
+
+  async ngOnInit() {
+    this.route.params.subscribe(async (params: any) => {
+      this.activeOrganizationId = params.organizationId;
+      await this.load();
+    });
+  }
+
+  async load() {
+    const orgs = await this.organizationService.getAll();
+    this.organizations = orgs
+      .filter((org) => org.canAccessAdminView)
+      .sort(Utils.getSortFunction(this.i18nService, "name"));
+
+    this.loaded = true;
+  }
```

Comment:
If we change to using inputs, this could be `ngOnChanges`.

Reply:
The input is only used for the active org, and that flows through the input binding automatically.
This logic only needs to be run when the component is first loaded.

Reply:
Marking this as resolved because it'll be in a separate PR